contracts/Zap.sol

```diff
@@ -236,6 +237,162 @@ contract Zap is AccessControlDefended {
+            poolId = 1;
+            // idx=0
+        }
+
+        (_lp, _fee) = ibbtcToCurveLP(2, amount);
+        _ren = pools[2].deposit.calc_withdraw_one_coin(_lp, 1);
+        if (_ren > renAmount) {
+            renAmount = _ren;
+            fee = _fee;
+            poolId = 2;
+            idx = 1;
+        }
+    }
+
+    /**
+    * @notice Calculate the most optimal route and expected wbtc amount when redeeming ibbtc.
+    * @dev Use returned params poolId, idx and wbtc in the call to redeem(...)
+           The last param `minOut` in redeem(...) should be a bit less than the returned wbtc value.
+           For instance 0.2% - 1% lesser depending on slippage tolerange.
+    * @param amount ibbtc amount
+    * @return poolId 0=crvRenWBTC, 1=crvRenWSBTC, 2=tbtc-sbtcCrv, 3=byvwbtc
+    * @return idx Index of the supported token in the curve pool (poolId)
+    * @return wBTCAmount Expected wbtc. Not for precise calculations. Doesn't factor in fee charged by the curve pool
+    * @return fee Fee being charged by ibbtc system. Denominated in corresponding sett token
+    */
+    function calcRedeemInWbtc(uint amount) public view returns(uint poolId, uint idx, uint wBTCAmount, uint fee) {
+        uint _lp;
+        uint _fee;
+        uint _wbtc;
+
+        // poolId=0, idx=0
+        (_lp, fee) = ibbtcToCurveLP(0, amount);
+        wBTCAmount = pools[0].deposit.calc_withdraw_one_coin(_lp, 1);
+        idx = 1;
+
+        (_lp, _fee) = ibbtcToCurveLP(1, amount);
+        _wbtc = pools[1].deposit.calc_withdraw_one_coin(_lp, 1);
+        if (_wbtc > wBTCAmount) {
+            wBTCAmount = _wbtc;
+            fee = _fee;
+            poolId = 1;
+            // idx=1
+        }
+
+        (_lp, _fee) = ibbtcToCurveLP(2, amount);
+        _wbtc = pools[2].deposit.calc_withdraw_one_coin(_lp, 2);
+        if (_wbtc > wBTCAmount) {
+            wBTCAmount = _wbtc;
+            fee = _fee;
+            poolId = 2;
+            idx = 2;
+        }
+
+        uint _byvWbtc;
+        uint _max;
+        (_byvWbtc,_fee,_max) = byvWbtcPeak.calcRedeem(amount);
+        if (amount <= _max) {
+            uint strategyFee = _byvWbtc.mul(pools[3].sett.withdrawalFee()).div(10000);
+            _wbtc = _byvWbtc.sub(strategyFee).mul(pools[3].sett.pricePerShare()).div(1e8);
+            if (_wbtc > wBTCAmount) {
+                wBTCAmount = _wbtc;
+                fee = _fee.add(strategyFee);
+                poolId = 3;
+            }
+        }
+    }
+
+    function ibbtcToCurveLP(uint poolId, uint bBtc) public view returns(uint lp, uint fee) {
+        uint sett;
+        uint max;
+        (sett,fee,max) = settPeak.calcRedeem(poolId, bBtc);
+        Pool memory pool = pools[poolId];
+        if (bBtc > max) {
+            return (0,fee);
+        } else {
+            // pesimistically charge 0.5% on the withdrawal.
+            // Actual fee might be lesser if the vault keeps keeps a buffer
+            uint strategyFee = sett.mul(50).div(1000);
```

Comment:
It's possible to be more intelligent with the fee
1. Check how many LP tokens are sitting in the buffer in the vault.
2. `controller.strategy.withdrawalFee`
Do we need to do so? or a default 0.5% ok because this function is just indicative anyway; not used in actual withdrawal.

Reply:
Wouldn't it make sense to fetch the fee dynamically? Even though the function is just indicative it will be used to generate the parameters for the actual redeem. I wonder how these parameters and, therefore the redeem, be affected if the fee changes but the estimation doesn't.

Reply:
Made the change :)